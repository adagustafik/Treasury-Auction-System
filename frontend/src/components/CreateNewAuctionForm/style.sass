.create-auction-container
    display: flex
    flex-flow: row wrap
    justify-content: center

.create-auction-col
    margin: 0 25px 0 0

.create-auction-container label
    margin: 10px 0 0 0

.terms-and-submit-container
    margin-top: 5px
    display: flex
    flex-flow: column wrap
    width: 100%

.terms-and-submit-container button
    margin: 10px 25px 0 0
    width:150px
    
.terms-and-submit-container textarea
    height: 100px 