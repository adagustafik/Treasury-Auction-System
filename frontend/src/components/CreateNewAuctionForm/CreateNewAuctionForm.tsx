import React from "react";
import { Button, Form } from "react-bootstrap";
import "./style.sass";

export function CreateNewAuctionForm() {
  return (
    <div className="create-auction-form">
      <Form className="create-auction-container">
        <div className="create-auction-col">
          <Form.Label>Auctions date</Form.Label>
          <Form.Control type="date" />
          <Form.Label>Issue date</Form.Label>
          <Form.Control type="date" />
          <Form.Label>Maturity date</Form.Label>
          <Form.Control type="date" />
          <Form.Label>Non-competitive bid closing date</Form.Label>
          <Form.Control type="date" />
          <Form.Label>Competitive bid closing date</Form.Label>
          <Form.Control type="date" />
        </div>
        <div className="create-auction-col">
          <Form.Label>Amount offered</Form.Label>
          <Form.Control type="text" />
          <Form.Label>Rate</Form.Label>
          <Form.Control type="text" />
          <Form.Label>Yield</Form.Label>
          <Form.Control type="text" />
          <Form.Label>Discount margin</Form.Label>
          <Form.Control type="text" />
        </div>
        <div className="terms-and-submit-container">
          <Form.Label>Terms and conditions</Form.Label>
          <Form.Control as="textarea" placeholder="terms.." />
          <Button type="submit">Create auction</Button>
        </div>
      </Form>
    </div>
  );
}
