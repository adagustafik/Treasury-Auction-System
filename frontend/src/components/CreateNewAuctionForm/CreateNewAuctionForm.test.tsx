import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { CreateNewAuctionForm } from "./CreateNewAuctionForm";

it("renders the CreatAuctionForm correctly", () => {
  const createAuctionForm = renderer
    .create(
      <BrowserRouter>
        <CreateNewAuctionForm />
      </BrowserRouter>
    )
    .toJSON();
  expect(createAuctionForm).toMatchSnapshot();
});
