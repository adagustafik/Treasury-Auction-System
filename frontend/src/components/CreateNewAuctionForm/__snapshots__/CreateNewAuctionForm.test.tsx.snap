// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CreatAuctionForm correctly 1`] = `
<div
  className="create-auction-form"
>
  <form
    className="create-auction-container"
  >
    <div
      className="create-auction-col"
    >
      <label
        className="form-label"
      >
        Auctions date
      </label>
      <input
        className="form-control"
        type="date"
      />
      <label
        className="form-label"
      >
        Issue date
      </label>
      <input
        className="form-control"
        type="date"
      />
      <label
        className="form-label"
      >
        Maturity date
      </label>
      <input
        className="form-control"
        type="date"
      />
      <label
        className="form-label"
      >
        Non-competitive bid closing date
      </label>
      <input
        className="form-control"
        type="date"
      />
      <label
        className="form-label"
      >
        Competitive bid closing date
      </label>
      <input
        className="form-control"
        type="date"
      />
    </div>
    <div
      className="create-auction-col"
    >
      <label
        className="form-label"
      >
        Amount offered
      </label>
      <input
        className="form-control"
        type="text"
      />
      <label
        className="form-label"
      >
        Rate
      </label>
      <input
        className="form-control"
        type="text"
      />
      <label
        className="form-label"
      >
        Yield
      </label>
      <input
        className="form-control"
        type="text"
      />
      <label
        className="form-label"
      >
        Discount margin
      </label>
      <input
        className="form-control"
        type="text"
      />
    </div>
    <div
      className="terms-and-submit-container"
    >
      <label
        className="form-label"
      >
        Terms and conditions
      </label>
      <textarea
        className="form-control"
        placeholder="terms.."
      />
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Create auction
      </button>
    </div>
  </form>
</div>
`;
